### DISCUSSION TITLE
Unexpected labels found (they should be expected)

### DISCUSSION THREAD

**Discussion by jsantiagogonzalez** [MIC-DKFZ/nnUNet#948] [Q&A] [answered]
Hi!

I'm getting started with nnUNet and when trying to preprocess my first dataset this happens:
The estructures of my data is:

nnUNet_raw_data_base/nnUNet_raw_data/Task001_MS
├── dataset.json
├── imagesTr
│ ├── MS_001_0000.nii.gz
│ ├── MS_001_0001.nii.gz
│ ├── ...
└── labelsTr
├── MS_001.nii.gz
├── MS_002.nii.gz
├── ...

My dataset.json is:

"name": "MS",
"description": "ISIB_CHALLENGE",
"reference": "Santiago González, GIB, GTB, UPM, HURyC",
"licence":"CC-BY-SA 4.0",
"relase":"1.0 19/02/2022",
"tensorImageSize": "3D",
"modality": {
"0000": "FLAIR",
"0001": "T1"
},
"labels": {
"0000":"BACKGROUND"
},
"numTraining": 32,
"numTest": 16,
"training":[{"image":"./imagesTr/MS_001.nii.gz","label":"./labelsTr/MS_001}...],
"test": []
}

and my environment variables are set as follows:
export nnUNet_raw_data_base="/home/alumno/TFG_SANTIAGO_GONZALEZ/nnUNet_raw_data_base"
export nnUNet_preprocessed="/home/alumno/TFG_SANTIAGO_GONZALEZ/nnUNet_preprocessed"
export RESULTS_FOLDER="home/alumno/TFG_SANTIAGO_GONZALEZ/nnUNet_trained_models"

But I get the following error when using the command: nnUNet_plan_and_preprocess -t 001 --verify_dataset_integrity:

Unexpected labels found in file /home/alumno/TFG_SANTIAGO_GONZALEZ/nnUNet_raw_data_base/nnUNet_raw_data/Task001_MS/labelsTr/MS_001.nii.gz. Found these unexpected values (they should not be there) [1]
....

(For every single image)

I don't know how to deal with this error please help.

Thank you!

**Comment by FabianIsensee** [comment, marked as answer]
You only declare the background label in the dataset.json. 

It needs to be 

"labels": {
"0":"BACKGROUND",
"1":"something",
"2":"something_else"
}

etc

**Reply by FabianIsensee** [reply]
Ah yeah it expects 'background' in lower case :upside_down_face:  my bad

**Reply by GMAvaliani** [reply]
Dear Fabian,
Can you help with Unexpected Labels in nnUNet.

I am currently working on setting up nnUNet and have encountered the same issue. When running a script, specifically:

C:\УИИ\Internship- модель nnU-Net\nnUNet-2.0\nnunetv2\new_try\nnUNet_raw>nnUNetv2_plan_and_preprocess -d 888 --verify_dataset_integrity

I am presented with the following error message:

Fingerprint extraction...
Dataset888_Lung
Using <class 'nnunetv2.imageio.simpleitk_reader_writer.SimpleITKIO'> as reader/writer
Error: Unexpected labels found in file C:/УИИ/Internship/- модель nnU-Net/nnUNet-2.0/- DATASET/nnUNet_raw\Dataset888_Lung\labelsTr\LUNG_001_0033.nii.gz.
Expected: [0, 1]
Found: [-1024. -1023. -1022. ... 1306. 1329. 1353.]

for example (attached) one of the slices has a Number of zeros: 35070748
![noname](https://github.com/MIC-DKFZ/nnUNet/assets/74610889/6242d169-e526-4168-97e1-af76f9ad4e91)

**Reply by GMAvaliani** [reply]
looks like Housefield units, but it's not clear why they're showing up here

**Reply by hdnminh** [reply]
Hi @GMAvaliani, have you fixed that bug? I was meeting this bug and have no idea to solve it too.
